### PR TITLE
Adding support for Autoclass 2.1

### PR DESCRIPTION
**Pull request by thomasmaclean**
Adds the `terminal_storage_class` field to autoclass configurations for `google_storage_bucket`.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
storage: added `terminal_storage_class` to the `autoclass` field in `google_storage_bucket` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb [hunk 1]
```diff
@@ -76,12 +78,13 @@ func TestAccStorageBucket_basicWithAutoclass(t *testing.T) {
...
 			{
-				Config: testAccStorageBucket_basicWithAutoclass(bucketName, false),
+				Config: testAccStorageBucket_basicWithAutoclass_update(bucketName, true),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(
-						"google_storage_bucket.bucket", "force_destroy", "false"),
+					testAccCheckStorageBucketExists(
+						t, "google_storage_bucket.bucket", bucketName, &updated),
+					testAccCheckStorageBucketWasUpdated(&updated, &bucket),
+					testAccCheckAutoclassTerminalStorageClass("ARCHIVE", &updated),
```
Comment (melinath) [inline mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:87]: Generally speaking, checks are only needed if the thing that you're testing is not returned by the API - for example, force_destroy is a Terraform-side flag, not an API field. Fields like terminal_storage_class will generally get checked during the ImportState step. Just want to double-check that this needs to be handled as a check?

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb [hunk 2]
```diff
@@ -76,12 +78,13 @@ func TestAccStorageBucket_basicWithAutoclass(t *testing.T) {
...
-			// Autoclass is ForceNew, so this destroys & recreates, but does test the explicitly disabled config
 			{
-				Config: testAccStorageBucket_basicWithAutoclass(bucketName, false),
+				Config: testAccStorageBucket_basicWithAutoclass_update(bucketName, true),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(
-						"google_storage_bucket.bucket", "force_destroy", "false"),
+					testAccCheckStorageBucketExists(
+						t, "google_storage_bucket.bucket", bucketName, &updated),
+					testAccCheckStorageBucketWasUpdated(&updated, &bucket),
```
Comment (melinath) [inline mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:86]: Haven't previously seen this as a solution for checking whether a delete happened - interesting! Seems reasonable.

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb [hunk 3]
```diff
@@ -76,12 +78,13 @@ func TestAccStorageBucket_basicWithAutoclass(t *testing.T) {
 				ImportStateVerify:       true,
 				ImportStateVerifyIgnore: []string{"force_destroy"},
 			},
-			// Autoclass is ForceNew, so this destroys & recreates, but does test the explicitly disabled config
 			{
-				Config: testAccStorageBucket_basicWithAutoclass(bucketName, false),
```
Comment (melinath) [inline mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:81]: What would you think about also keeping this config around to make sure that disabling works as expected?

**Review by melinath — CHANGES_REQUESTED** [review]
This seems straightforward; I just have a couple questions.

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb (hunk 2 above)
Comment (thomasmaclean) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:86]: Can't take credit for this one... it's found elsewhere in the file (the storage class tests I believe).

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb (hunk 3 above)
Comment (thomasmaclean) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:81]: Good call.

File: mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb (hunk 1 above)
Comment (thomasmaclean) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb:87]: I set the field as `Computed` and wanted to be really clear that it was being set.

Comment (thomasmaclean) [reply mmv1/third_party/terraform/services/storage/resource_storage_bucket_test.go.erb]: I did this more for my own verification while I was testing, so I'm just going to delete it.

**Review by melinath — APPROVED** [review]

**Pull request merged**